src/main/java/com/example/tournamentMaker/excel/football/FootballLeagueWriterService.java

```diff
@@ -0,0 +1,187 @@
+    private static final String SHEET_NAME = "League";
+    private final FootballStatisticWriterService footballStatisticWriterService;
+    private final Workbook workbook = new XSSFWorkbook();
+    private final Sheet sheet = workbook.createSheet(SHEET_NAME);
+    private static final int TEAM_COLUMN_EXCEL_WIDTH = 26;
+    private static final int COLUMN_WITH_NUMBER_EXCEL_WIDTH = 5;
+    private static final int LAST_5_COLUMN_EXCEL_WIDTH = 12;
+    private static final int STATISTICS_COLUMN_START = 9;
+    private static final int STATISTICS_ROW_START = 0;
+    private final StatisticService statisticService = new StatisticService();
+
+    @Override
+    public boolean writeTournamentInformation(Tournament tournament) {
+        CellStyle borderCellStyle = ExcelUtil.createCellStyle(workbook, IndexedColors.GOLD.getIndex());
+        CellStyle fillingTableCellStyle = ExcelUtil.createCellStyle(workbook, IndexedColors.LEMON_CHIFFON.getIndex());
+
+        List<Statistics> statistics = tournament.getTeamList().stream()
+                .map(team -> team.getStatistics())
+                .sorted(Comparator.comparingInt(Statistics::getPoints).reversed())
+                .collect(Collectors.toList());
+
+        int rowStart = 0;
+        int colStart = 0;
+        createColumnHeaders(rowStart, colStart, borderCellStyle);
+        rowStart++;
+
+        Font greenFont = workbook.createFont();
+        greenFont.setColor(IndexedColors.GREEN.getIndex());
+        Font redFont = workbook.createFont();
+        redFont.setColor(IndexedColors.RED.getIndex());
+        Font blueFont = workbook.createFont();
+        blueFont.setColor(IndexedColors.BLUE.getIndex());
+
+        for (int i = 0; i < statistics.size(); i++) {
+            Statistics stats = statistics.get(i);
+            Row row = sheet.createRow(rowStart);
+
+            Cell cell = row.createCell(colStart);
+            cell.setCellStyle(fillingTableCellStyle);
+            int place = i + 1;
+            cell.setCellValue(place);
+            colStart++;
+
+            cell = row.createCell(colStart);
+            cell.setCellStyle(fillingTableCellStyle);
+            cell.setCellValue(stats.getTeam().getName());
+            colStart++;
+
+            cell = row.createCell(colStart);
+            cell.setCellStyle(fillingTableCellStyle);
+            cell.setCellValue(statisticService.getGamesCount(stats));
+            colStart++;
+
+            cell = row.createCell(colStart);
+            cell.setCellStyle(fillingTableCellStyle);
+            cell.setCellValue(stats.getCountWins());
+            colStart++;
+
+            cell = row.createCell(colStart);
+            cell.setCellStyle(fillingTableCellStyle);
+            cell.setCellValue(stats.getCountDraws());
+            colStart++;
+
+            cell = row.createCell(colStart);
+            cell.setCellStyle(fillingTableCellStyle);
+            cell.setCellValue(stats.getCountLoses());
+            colStart++;
```

Comment:
to wyglada zajebiscie dziwnie. Jak jakis ultra beznadziejny design kodu. W obrebie jednej petli walisz colStart++ nie wiadomo ile razy. Btw na cholere Ci ta zmienna ? co od niej zalezy pozniej

Reply:
A juz widze, ale taka metoda to jest jakas masakra, nie moge na to patrzec